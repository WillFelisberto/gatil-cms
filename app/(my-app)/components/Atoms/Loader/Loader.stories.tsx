import type { Meta, StoryObj } from '@storybook/react';

import { Loader } from './Loader';

const meta: Meta<typeof Loader> = {
  title: 'Atoms/Loader',
  component: Loader,
  tags: ['autodocs']
};

export default meta;
type Story = StoryObj<typeof Loader>;

export const Default: Story = {
  args: {}
};

export const WithText: Story = {
  args: {
    text: 'Carregando gatinhos ador√°veis...'
  }
};
