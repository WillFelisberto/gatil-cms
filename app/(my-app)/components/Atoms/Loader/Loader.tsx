import { PawPrint } from 'lucide-react';

type LoaderProps = {
  text?: string;
};

export const Loader = ({ text }: LoaderProps) => {
  return (
    <div className="flex flex-col items-center justify-center h-[70vh] gap-4">
      <div className="animate-bounce text-[#013274]">
        <PawPrint className="w-16 h-16" data-testid="paw-icon" />
      </div>
      {text && <p className="text-[#013274] font-semibold text-lg text-center">{text}</p>}
    </div>
  );
};
