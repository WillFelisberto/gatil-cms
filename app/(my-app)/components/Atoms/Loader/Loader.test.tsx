import { render, screen } from '@/tests/test-utils';

import { Loader } from './Loader';

describe('<Loader />', () => {
  it('renders the icon (PawPrint)', () => {
    const { container } = render(<Loader />);
    expect(screen.getByTestId('paw-icon')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('does not render text when "text" prop is not provided', () => {
    render(<Loader />);
    const paragraph = screen.queryByText(/.+/);
    expect(paragraph).not.toBeInTheDocument();
  });

  it('renders text when "text" prop is provided', () => {
    const testText = 'Carregando gatinhos...';
    render(<Loader text={testText} />);
    expect(screen.getByText(testText)).toBeInTheDocument();
  });
});
