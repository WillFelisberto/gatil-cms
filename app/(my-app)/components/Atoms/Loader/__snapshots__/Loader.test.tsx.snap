// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders the icon (PawPrint) 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center h-[70vh] gap-4"
  >
    <div
      class="animate-bounce text-[#013274]"
    >
      <svg
        class="lucide lucide-paw-print w-16 h-16"
        data-testid="paw-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="4"
          r="2"
        />
        <circle
          cx="18"
          cy="8"
          r="2"
        />
        <circle
          cx="20"
          cy="16"
          r="2"
        />
        <path
          d="M9 10a5 5 0 0 1 5 5v3.5a3.5 3.5 0 0 1-6.84 1.045Q6.52 17.48 4.46 16.84A3.5 3.5 0 0 1 5.5 10Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
